modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts

```diff
@@ -361,6 +389,92 @@ export class LinkModalDialog
         return this.mediaOptionRadioFormItem;
     }
 
+    private createUrlFormItem(textId: string, textLabel: string): FormItem {
+        const getUrl = () => this.isUrl() ? this.link : '';
+
+        const urlFormItem: FormItem = this.createFormItemWithPostponedValue(textId, textLabel, getUrl);
+        const urlInput = <TextInput>urlFormItem.getInput();
+
+        urlInput.onRendered(() => {
+            const urlValue = urlInput.getValue();
+            const usedProtocol = this.getUsedProtocolFromValue(urlValue);
+            const actionTitle = urlValue ? usedProtocol.title : 'Https';
+            const action = this.protocolsDropdownButton.getMenuActions().find(action => action.getLabel() === actionTitle);
+            if (action) { action.execute(); }
+        });
+
+        urlInput.onValueChanged((event: ValueChangedEvent) => {
+            const urlValue = event.getNewValue();
+            const usedProtocol = this.getUsedProtocolFromValue(urlValue);
+
+            this.protocolsDropdownButton.getMenuItems().forEach(menuItem => {
+                menuItem.removeClass('menu-item-selected');
+                if (menuItem.getEl().getInnerHtml() === usedProtocol.title) {
+                    menuItem.addClass('menu-item-selected');
+                }
+            });
+        });
+
+        urlFormItem.getLabel().addClass('required');
+
+        return urlFormItem;
+    }
+
+    private createProtocolsDropdownButton(textInput: TextInput): MenuButton {
+        const protocolsDropdownButton = new MenuButton(new Action('Type'));
+
+        const actions = this.urlProtocols.map(({title, prefix}) => {
+            const action = new Action(title);
+
+            action.onExecuted(() => {
+                const urlValue = textInput.getValue();
+                const usedProtocol = this.getUsedProtocolFromValue(urlValue);
+
+                if (!prefix) {
+                    if (!usedProtocol.prefix) {
+                        textInput.setValue(urlValue);
```

Comment:
Instead of repeating `textInput.setValue()` 4 times, declare a string variable called "`newUrlValue`", assign a new value to it and then call `textInput.setValue(newUrlValue)` after all the `if`s.